packages/flutter/lib/src/rendering/object.dart

```diff
@@ -1088,6 +1088,14 @@ base class PipelineOwner with DiagnosticableTreeMixin {
   bool _shouldMergeDirtyNodes = false;
   List<RenderObject> _nodesNeedingLayout = <RenderObject>[];
 
+  /// The [RenderObject]s which need to be laid out in the next [flushLayout] pass.
+  ///
+  /// [RenderObject]s with [RenderObject.isRepaintBoundary] are added to this list
+  /// when they are marked for layout. Subclasses of [PipelineOwner] may use this list
+  /// to invalidate caches or otherwise make performance optimizations.
+  @protected
+  List<RenderObject> get nodesNeedingLayout => _nodesNeedingLayout;
```

Comment:
I would expose these as Iterables so that its more obvious they are intended to be read only. I guess in theory adding nodes to this list shouldn't cause bugs but I wouldn't want to test that.

Reply:
Great idea, thanks! Done.